adapters/nft-key/index.js

```diff
@@ -0,0 +1,164 @@
+            address indexed erc721Address,
+            uint256 indexed tokenId,
+            Bid bid
+        );
+        event TokenBidWithdrawn(
+            address indexed erc721Address,
+            uint256 indexed tokenId,
+            Bid bid
+        );
+        event TokenBought(
+            address indexed erc721Address,
+            uint256 indexed tokenId,
+            address indexed buyer,
+            Listing listing,
+            uint256 serviceFee,
+            uint256 royaltyFee
+        );
+        event TokenBidAccepted(
+            address indexed erc721Address,
+            uint256 indexed tokenId,
+            address indexed seller,
+            Bid bid,
+            uint256 serviceFee,
+            uint256 royaltyFee];
+        this.pathToAbi = path.join(__dirname, "./abi.json");
+        this.range = 500;
+        this.chunkSize = 6;
+        this.sdk = new Avalanche(this);
+    }
+
+    run = async () => {
+        const s = await this.getSymbol();
+        this.sdk = this.loadSdk();
+        this.symbol = s;
+        await this.sdk.run();
+    };
+
+    loadSdk = () => {
+        return new Avalanche(this);
+    };
+    getSymbol = async () => {
+        const resp = await axios.get(
+            `${URL}/token-metadata?key=${KEY}&token_address=${this.token}&protocol=${this.protocol}`,
+            {
+                headers: {
+                    "User-Agent":
+                        "Mozilla/5.0 (Macintosh; Intel Mac OS X 11_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4619.141 Safari/537.36",
+                },
+            },
+        );
+        const symbol = resp.data;
+        return symbol;
+    };
+    getPrice = async timestamp => {
+        const resp = await axios.get(
+            `${URL}/token-price?key=${KEY}&token_address=${this.token}&protocol=${this.protocol}&timestamp=${timestamp}`,
+            {
+                headers: {
+                    "User-Agent":
+                        "Mozilla/5.0 (Macintosh; Intel Mac OS X 11_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4619.141 Safari/537.36",
+                },
+            },
+        );
+        return resp.data;
+    };
+    stop = async () => {
+        this.sdk.stop();
+    };
+
+    getBuyer = async event => {
+        const buyer = event.returnValues.owner;
+        if (event.event === "Offer, OfferCreated, OfferCancelled, OfferAccepted") {
+            const txReceipt = await this.sdk.getTransactionReceipt(event.transactionHash);
+            if (txReceipt === null) {
+                return null;
+            }
+            return txReceipt.from;
+        }
+        return buyer;
+    };
+
+    process = async event => {
+        const block = await this.sdk.getBlock(event.blockNumber);
+        const timestamp = moment.unix(block.timestamp).utc();
+        const po = await this.getPrice(block.timestamp);
+        const nativePrice = new BigNumber(event.returnValues.cost).dividedBy(10 ** this.symbol.decimals);
```

Comment:
Your event doesn't have `cost` property.

Reply:
Thank you for pointing this out.